### PR TITLE
Speed up initcode validation

### PR DESCRIPTION
**Pull request by shemnon**
## PR description

Speedup initcode validation by
* optimizing the jumpdest loop
* not caching initcode in the code cache (but keep codecache for normal contracts).

Signed-off-by: Danno Ferrin <danno.ferrin@swirldslabs.com>

## Documentation

- [ ] I thought about documentation and added the `doc-change-required` label to this PR if
    [updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).

## Changelog

- [ ] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: evm/src/main/java/org/hyperledger/besu/evm/code/CodeV0.java [hunk 1]
```diff
@@ -155,15 +156,141 @@ long[] calculateJumpDests() {
...
+            case 0x7e:
+              i += 31;
+              j += 31;
+              break;
+            case 0x7f:
+              i += 32;
+              j += 32;
+              break;
+            default:
+          }
```
Comment (ahamlat) [inline evm/src/main/java/org/hyperledger/besu/evm/code/CodeV0.java:293]: Do you confirm this is not related to not cache initcode ? 
If it is not related, maybe change the PR title or create another PR

**Review by ahamlat — COMMENTED** [review]
Nice PR, I added one comment on JumpDest

File: evm/src/main/java/org/hyperledger/besu/evm/code/CodeV0.java (hunk 1 above)
Comment (shemnon) [reply evm/src/main/java/org/hyperledger/besu/evm/code/CodeV0.java:293]: Changed name and comments

**Review by ahamlat — APPROVED** [review]
